rcl/include/rcl/security.h

```diff
@@ -0,0 +1,123 @@
+#endif
+
+#ifndef ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME
+# define ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME "ROS_SECURITY_ROOT_DIRECTORY"
+#endif
+
+#ifndef ROS_SECURITY_STRATEGY_VAR_NAME
+# define ROS_SECURITY_STRATEGY_VAR_NAME "ROS_SECURITY_STRATEGY"
+#endif
+
+#ifndef ROS_SECURITY_ENABLE_VAR_NAME
+# define ROS_SECURITY_ENABLE_VAR_NAME "ROS_SECURITY_ENABLE"
+#endif
+
+/// Initialize security options from values in the environment variables and given names.
+/**
+ * Initialize the given security options based on the environment.
+ * For more details:
+ *  \sa rcl_security_enabled
+ *  \sa rcl_get_enforcement_policy
+ *  \sa rcl_get_secure_root
+ *
+ * \param[in] name name used to find the securiy root path.
+ * \param[in] allocator used to do allocations.
+ * \param[out] security_options security options that will be configured according to
+ *  the environment.
+ */
+RCL_PUBLIC
+rcl_ret_t
+rcl_get_security_options_from_environment(
+  const char * name,
+  const rcutils_allocator_t * allocator,
+  rmw_security_options_t * security_options);
+
+/// Check if security has to be used, according to the environment.
+/**
+ * If `ROS_SECURITY_ENABLE` environment variable is set to "true", `use_security` will be set to
+ * true.
+ *
+ * \param[out] use_security Must not be NULL.
+ * \returns RCL_RET_INVALID_ARGUMENT if an argument is not valid, or
+ * \returns RCL_RET_ERROR if an unexpected error happened, or
+ * \returns RCL_RET_OK.
+ */
+RCL_PUBLIC
+rcl_ret_t
+rcl_security_enabled(bool * use_security);
+
+/// Get security enforcement policy from the environment.
+/**
+ * Sets `policy` based on the value of `ROS_SECURITY_STRATEGY` environment variable.
+ * If `ROS_SECURITY_STRATEGY` is "Enforce", `policy` will be `RMW_SECURITY_ENFORCEMENT_ENFORCE`.
+ * If not, `policy` will be `RMW_SECURITY_ENFORCEMENT_PERMISSIVE`.
+ *
+ * \param[out] policy Must not be NULL.
+ * \returns RCL_RET_INVALID_ARGUMENT if an argument is not valid, or
+ * \returns RCL_RET_ERROR if an unexpected error happened, or
+ * \returns RCL_RET_OK.
+ */
+RCL_PUBLIC
+rcl_ret_t
+rcl_get_enforcement_policy(rmw_security_enforcement_policy_t * policy);
+
+/// Return the secure root given a security context name.
+/**
+ * Return the security directory associated with the security context name.
+ *
+ * The value of the environment variable `ROS_SECURITY_ROOT_DIRECTORY` is used as a root.
+ * The specific directory to be used, is found from that root using the `name` passed.
+ * E.g. for a context named "/a/b/c" and root "/r", the secure root path will be
+ * "/r/a/b/c", where the delimiter "/" is native for target file system (e.g. "\\" for _WIN32).
+ *
+ * However, this expansion can be overridden by setting the secure directory override environment
+ * (`ROS_SECURITY_DIRECTORY_OVERRIDE`) variable, allowing users to explicitly specify the exact secure
+ * root directory to be utilized.
+ * Such an override is useful for where the security context is non-deterministic before runtime,
```

Comment:
@ivanpauno nit: 
```suggestion
 * Such an override is useful for applications where the security context is non-deterministic before runtime,
```

Reply:
Also, what do you mean by `non-deterministic before runtime`? Unknown?

Reply:
Again, this was taken from the old `security_directory.h`.
The original comment isn't clear to me.

Reply:
Used your suggestion in https://github.com/ros2/rcl/pull/515/commits/35ec36eb170bb5801c17f03637e2a9b97f13ded7.